x-pack/plugins/discover_enhanced/public/actions/explore_data/explore_data_context_menu_action.ts

```diff
@@ -42,9 +41,9 @@ export class ExploreDataContextMenuAction extends AbstractExploreDataAction<Embe
 
       const input = embeddable.getInput();
 
-      if (isTimeRange(input.timeRange) && !state.timeRange) state.timeRange = input.timeRange;
-      if (isQuery(input.query)) state.query = input.query;
-      if (isFilters(input.filters)) state.filters = [...input.filters, ...(state.filters || [])];
+      if (input.timeRange && !state.timeRange) state.timeRange = input.timeRange;
```

Comment:
why is it ok to remove these checks now?

Reply:
Because of this change:

![image](https://user-images.githubusercontent.com/9773803/90865808-4117d500-e393-11ea-85b9-c3686b7ca59d.png)

It moves those properties onto the base embeddable.

Reply:
so the expectation is that each embeddable would implement these checks itself?

Reply:
I'm not sure I completely understand the questions. Embeddables don't need to implement any checks.

Each embeddable will have optional `query`, `filters` and `timeRange` properties on its input. It is up to embeddable if it implements those fields.

All __consumers__ of the embeddable now need to do a simple check, like

```ts
if (embeddable.getInput().filters) ...
```

to know that embeddable has filters. Before that it was more tricky to know if embeddable has filters.